pkg/collector/corechecks/network-devices/cisco-sdwan/sdwan.go

```diff
@@ -0,0 +1,211 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package ciscosdwan implements NDM Cisco SD-WAN corecheck
+package ciscosdwan
+
+import (
+	"time"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/client"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/payload"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/report"
+	"github.com/DataDog/datadog-agent/pkg/snmp/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+const (
+	// CheckName is the name of the check
+	CheckName = "ciscosdwan"
+)
+
+// Configuration for the Cisco SD-WAN check
+type checkCfg struct {
+	Hostname    string `yaml:"hostname"`
+	Port        int    `yaml:"port"`
+	Username    string `yaml:"username"`
+	Password    string `yaml:"password"`
+	Namespace   string `yaml:"namespace"`
+	MaxAttempts int    `yaml:"max_attempts"`
+	MaxPages    int    `yaml:"max_pages"`
+	MaxCount    int    `yaml:"max_count"`
+	Lookback    int    `yaml:"lookback"`
+	UseHTTP     bool   `yaml:"use_http"`
+	Insecure    bool   `yaml:"insecure"`
+	CAFile      string `yaml:"ca_file"`
+	NoMetadata  bool   `yaml:"no_metadata"`
```

Comment:
I think negation in config parameters are harder to read usually, what about changing that to `send_ndm_metadata` ?
This field is sensitive as this is one that will trigger billing once Cisco SDWAN is GA

Reply:
Agree 100%, was trying to avoid having to use pointers since the default is true 😅
Updated in 877a6ad thanks